aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,392 @@
+
+## Typed clients
+
+Typed clients provide the same capabilities as named clients without the need to use strings as keys. This provides IntelliSense and compiler help when consuming clients. They also provide a single location to configure and interact with a particular `HttpClient`. For example, a single typed client might be used for a single backend endpoint and encapsulate all logic which deals with that endpoint. 
+
+A typed client is expected to accept a HttpClient via it's constructor. 
+
+```csharp
+public class GitHubService
+{
+    public HttpClient Client { get; private set; }
+
+    public GitHubService(HttpClient client)
+    {
+        client.BaseAddress = new Uri("https://api.github.com/");
+        client.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json"); // GitHub API versioning
+        client.DefaultRequestHeaders.Add("User-Agent", "HttpClientFactory-Sample"); // GitHub requires a user-agent
+
+        Client = client;
+    }
+}
+```
+In this example, only the configuration is moved into the typed client and the `HttpClient` is exposed directly as a property. It is also possible to define API specific methods which expose `HttpClient` functionality. 
+
+To register a typed client, the generic `AddHttpClient` extension method can be used, specifying the typed client class. 
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient<GitHubService>();
+    services.AddMvc();
+}
+```
+
+The typed client is registered as transient with the DI framework. The typed client can then be injected and consumed directly. 
+
+```csharp
+public class MyController : Controller
+{
+    private GitHubService _gitHubService;
+
+    public MyController(GitHubService gitHubService)
+    {
+        _gitHubService = gitHubService;
+    }
+
+    public IActionResult Index()
+    {
+        var result = await _gitHubService.Client.GetStringAsync("/orgs/octokit/repos");
+        return Ok(result);
+    }
+}
+```
+
+If preferred, the configuration for a typed client can be specified during registration, rather than in the constructor for the typed client.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient<GitHubService>(c =>
+    {
+        c.BaseAddress = new Uri("https://api.github.com/");
+
+        c.DefaultRequestHeaders.Add("Accept", "application/vnd.github.v3+json"); // GitHub API versioning
+        c.DefaultRequestHeaders.Add("User-Agent", "HttpClientFactory-Sample"); // GitHub requires a user-agent
+    });
+    services.AddMvc();
+}
+```
+
+It is possible to entirely encapsulate the `HttpClient` within a typed client. Rather than exposing it as a property, public methods can be provided which call the `HttpClient` instance internally.
```

Comment:
- it is --> it's
- Revisit this 2nd sentence. Not sure what it's trying to say. Maybe there's a word missing?